### DISCUSSION TITLE
Peloton Bike

### DISCUSSION THREAD

**Discussion by doudar** [doudar/SmartSpin2k#82] [Show and tell]
Just a note - I was finally able to make an insert and test this out on a friends Peloton Bike. The Peloton works *VERY* well with this as the resistance range on the bike is huge and the magnetic resistance is fantastic. I was able to get my full 1200w during a sprint with it.

If anyone is interested, you'll find the Peloton insert at https://github.com/doudar/SmartSpin2k/blob/master/Hardware/Inserts/Peloton_60mm.STL

**Comment by chaloney** [comment]
After printing a second part at a slightly larger size because the first one didn’t fit on the knob, I noticed that my peloton’s knob has seven indentations rather than six. It looks like if the opening had seven sides that were each 26mm long it would fit just loose enough to slip on but have only a little backlash

**Reply by doudar** [reply]
Crazy, I wonder I there are two different knobs for peloton then?

I'll design one to your specs and upload it today.

Thanks Matt!

**Reply by chaloney** [reply]
i bodged together a 3d model last night but was too tired to start the print before going to bed.  i made it all one piece with the cup as i only have the peloton to use with the smartspin.  the sides are 25.7 mm due to how MS 3d builder can make polygons but i think it will work  will print it tonight and let you know how it fits tomorrow
also, i used an elegoo mars resin printer to print everything but the strap and it all fits and prints well.

**Reply by doudar** [reply]
Nice! Love to see some pictures. 

I just uploaded your insert. You'll find it here: 
https://github.com/doudar/SmartSpin2k/blob/develop/Hardware/Inserts/Peloton_7_flat.STL

If you do get a chance to print it, please let me know if it works! 

Thanks!

**Reply by chaloney** [reply]
printed it and it engages the knob but only the top. im not sure if the resin print is shrinking some with curing - i will dig out the calipers tonight.   i have revised your model with a larger opening.  will be printing over night tonight after work and let you know how it fits.  i can send you the stl once i get it to the right size.

**Reply by doudar** [reply]
Cool, let me know how it goes. It's hard to get it exact because of material expansion & contraction during the printing process varies a bit between printers.

**Reply by chaloney** [reply]
new insert fits. i emailed the stl to smartspin2k@gmail.com

**Comment by chaloney** [comment]
finally have everything wired up.   my intent is to use a gymnasticon with the smartspin2k to essentially turn the peloton into a smartbike, so i did not bother with the power up/down buttons.  i am running trainerroad on my phone.  everything connects and appears to be working.  trainerroad is connected to the smartspin2k.  it shows power and cadence numbers properly.  trainerroad does not appear to be able to control the smartspin power, it just stays the same.  i can connect pin 19 on the esp to ground and make the resistance increase or pin 18 to ground and the resistance will decrease.  separately from that, the smartspin2k does not turn the resistance knob.  any suggestions?

**Reply by doudar** [reply]
It could be that we need to do some work with Trainer Road.

What mode are you using on trainer road?

If you can, please post a screenshot of the debugging information that has a line <-from app in it. 

Thanks!

**Comment by chaloney** [comment]
Something like this?

**Reply by chaloney** [reply]
![IMG_3695](https://user-images.githubusercontent.com/79726549/116269411-35638600-a74c-11eb-970e-390b0cf141a0.PNG)

**Reply by chaloney** [reply]
i am connected to the smartspin2k as a "smart trainer" in trainerroad, using ERG mode over bluetooth.  i saw in an earlier discussion ("compatibility questions") that erg mode was not yet working, is that still the case?

i redid the uart wiring using soldered connections to the resistor to be sure that wasn't the problem--no change
i tried using the trainerroad app on my laptop -- no change
also used the xert player app on my iphone -- same behavior.  

my esp32 has 3 separate ground pins so i connected to each individually and in all combinations of pairs or all three - no change.  continues to have no movement of the stepper motor except when i connect pin18 or 19 to ground (ie. "push" the shift up or down buttons)

**Reply by doudar** [reply]
So the screenshot above looks like no app is connected to the smartspin2k, or it (the app) isn't sending out a control point for some reason. You should see <--From App if it's receiving information.  

Are you able to try Zwift of Fulgaz just to see if the control works?

**Reply by chaloney** [reply]
Zwift seems to work.  The <-- From App lines appear with a incline number.  also could see the smartspin2k moving the resistance knob.  

zwift also works with its workouts in erg mode.  the screenshot below is from the start of one of a zwift workout.  the resistance was being adjusted to match the target
![Capture1](https://user-images.githubusercontent.com/79726549/116608526-4bff0e00-a901-11eb-98f6-aad980f31949.JPG)

**Reply by kadaan** [reply]
@chaloney I haven'y used TrainerRoad, but would gladly try it out on my IC4.  There doesn't seem to be any free trial,  but I see lots of mentions about 1 month free referrals.  Know the deal with that?

We know that Zwift, Kinetic, FulGaz all work.  Pretty sure Kinomap, RGT, Rouvy, and Sufferfest work as well.

**Reply by kadaan** [reply]
@chaloney Asking them for a developer account: https://www.trainerroad.com/forum/t/developer-account/58015. We will see what they say.

**Comment by chaloney** [comment]
Apparently each month a member pays for, they get a referral code for a free intro month for someone else. Here’s links for a couple I haven’t given out:

<EDIT>
Removed referrals
<EDIT>

**Reply by doudar** [reply]
Cool - we should be able to figure out what trainer road doesn't like about our BLE.

Thanks Matt

**Comment by chaloney** [comment]
I have an ant+ usb stick that I will be able to get this weekend. I will try it with ant+ next week.

**Reply by kadaan** [reply]
> I have an ant+ usb stick that I will be able to get this weekend. I will try it with ant+ next week.

@chaloney Unfortunately ANT+ won't work because we don't have an ANT+ chip on the microcontroller.  That is something we are looking at for the next version.

**Comment by chaloney** [comment]
Ok. I guess I was thinking of the gymnasticon then. Thanks. Now I won’t feel bad when I inevitably forget to bring it home this weekend ;)

**Reply by kadaan** [reply]
And won't waste any of your valuable weekend time...

**Reply by kadaan** [reply]
@chaloney TrainerRoad gave us an account.  I've deleted the referrals you posted so you can hopefully use them again if you want.

**Comment by chaloney** [comment]
That’s great. I saw the thread in the forum a few minutes ago. Thank you.

**Reply by doudar** [reply]
Great work guys!
